.github/workflows/release-azure-containerapp-v1.yaml

```diff
@@ -0,0 +1,376 @@
+on:
+  workflow_call:
+    inputs:
+      dockerfile_path:
+        type: string
+        required: false
+        default: './Dockerfile'
+        description: Dockerfile path
+      dockerfile_context:
+        type: string
+        required: false
+        default: '.'
+        description: Dockerfile build context
+      docker_image_name:
+        type: string
+        required: false
+        default: ${{ github.repository }}
+        description: Docker image name, defaults to repository name
+      docker_image_description:
+        type: string
+        required: true
+        description: A description to use as image label
+      docker_image_authors:
+        type: string
+        required: false
+        default: PagoPA
+        description: Authors names to use as image label
+      build_args:
+        description: List of build arguments to use for Dockerfile build, given in env=value format.
+        type: string
+        required: false
+      build_platforms:
+        type: string
+        required: false
+        default: linux/amd64
+        description: Image runtime platform, supports multiple comma-separated values
+      container_app:
+        type: string
+        required: true
+        description: Name of the container app to deploy
+      resource_group_name:
+        type: string
+        required: true
+        description: Name of the resource group of the container app
+      environment:
+        description: Environment where the image will be deployed
+        type: string
+        required: true
+      script_language:
+        type: string
+        default: 'typescript (npm)'
+        required: false
+        description: Define the language and package manager of the script to run to check for errors during canary deployment. Available only when Revision Mode is set to Multiple. Supports 'typescript (npm)', 'typescript (yarn)' values only.
+      script_name:
+        type: string
+        required: false
+        description: Script name to execute during canary deployment. When using npm, it refers to the script path; when using yarn, it is used as script name. Available only when Revision Mode is set to Multiple
+
+concurrency:
+  group: ${{ github.workflow }}-cd
+  cancel-in-progress: true
+
+env:
+  IMAGE_NAME: ${{ inputs.docker_image_name }}
+
+jobs:
+  build:
+    name: Build Docker Image
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write
+      attestations: write
+      packages: write
+
+    steps:
+
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        name: Checkout
```

Comment:
thought: what about adding `fetch-depth:1`?

Reply:
Time saving is minimal to me

Reply:
I guess it depends on the repo size and history